DataFormats/ParticleFlowReco/interface/PFBlockElementTrack.h

```diff
@@ -50,12 +50,12 @@ namespace reco {
 
     /// check if the track is secondary
     bool isSecondary() const override {
-      return trackType(T_FROM_DISP) || trackType(T_FROM_GAMMACONV) || trackType(T_FROM_V0);
+      return trackType(T_FROM_DISP) | trackType(T_FROM_GAMMACONV) | trackType(T_FROM_V0);
     }
 
     bool isPrimary() const override { return trackType(T_TO_DISP); }
 
-    bool isLinkedToDisplacedVertex() const override { return isSecondary() || isPrimary(); }
+    bool isLinkedToDisplacedVertex() const override { return isSecondary() | isPrimary(); }
```

Comment:
here and in `isPrimary` and `isSecondary` can we get bitmasks instead?

```suggestion
    static constexpr unsigned int kPrimaryMask = 1 << T_TO_DISP;
    static constexpr unsigned int kSecondaryMask = (1 << T_FROM_DISP) | (1 << T_FROM_GAMMACONV) | (1 << T_FROM_V0);
    static constexpr unsigned int kLinkedToDisplacedVertexMask = kPrimaryMask | kSecondaryMask;
    bool isSecondary() const override { return trackType_ & kSecondaryMask; } 
    bool isPrimary() const override { return trackType_ & kPrimaryMask; }
    bool isLinkedToDisplacedVertex() const override { return trackType_ & kLinkedToDisplacedVertexMask; }
```

it's worth testing the performance  though

Reply:
Thanks for the suggestion! I have pushed these changes. The resulting performance improvement is non-negligible:

https://msaunder.web.cern.ch/msaunder/cgi-bin/igprof-navigator/gitIssues/33083/step3_23434_21_bitmasked/3400